ESQL: tests for LOOKUP JOIN with non-unique join keys

alex-spies opened the pull request:
Add a csv dataset and tests for `LOOKUP JOIN` where the join keys are not unique. In particular, add tests that include MVs and nulls to see how `LOOKUP JOIN` treats these.

alex-spies committed:
Add tests with non-unique key in LU index

alex-spies committed:
Merge remote-tracking branch 'upstream/main' into lookup-join-non-unique-tests

alex-spies committed:
Denormalize data a little + add a field

alex-spies committed:
Merge remote-tracking branch 'upstream/main' into lookup-join-non-unique-tests

elasticsearchmachine commented:
Pinging @elastic/es-analytical-engine (Team:Analytics)

craigtaverner commented on x-pack/plugin/esql/qa/testFixtures/src/main/resources/lookup-join.csv-spec:
```diff
@@ -325,3 +324,80 @@ count:long | type:keyword
 1          | Disconnected
 ;
+
+nonUniqueLeftKeyOnTheDataNode
```
I've been grouping all tests that work with employees-join-languages at the top of the file, and joins against clientips and message-types following in blocks of tests. Perhaps we can move this to before the clientips tests, just after the languages tests, so it aligns with my PR (and therefore reduces conflicts too).

craigtaverner reviewed (approved):
LGTM, but I would prefer the tests got moved to before the clientips tests to reduce conflict with my PR.

craigtaverner committed:
Merge branch 'main' into lookup-join-non-unique-tests

alex-spies committed:
Move new tests up

Just after the tests with the languages_lookup index, for better
organization of the tests.

alex-spies committed:
Merge remote-tracking branch 'upstream/main' into lookup-join-non-unique-tests

alex-spies committed:
Merge remote-tracking branch 'upstream/main' into lookup-join-non-unique-tests

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.x](https://github.com/elastic/elasticsearch/pull/118661)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/118661">](https://github.com/elastic/elasticsearch/pull/118661) |
